clang/runtime/dpct-rt/include/dpct/graph.hpp

```diff
@@ -22,30 +22,53 @@ typedef sycl::ext::oneapi::experimental::command_graph<
     sycl::ext::oneapi::experimental::graph_state::executable>
     *command_graph_exec_ptr;
 
-static inline std::unordered_map<sycl::queue *, command_graph_ptr>
-    queue_graph_map;
-
-static bool command_graph_begin_recording(sycl::queue *queue_ptr) {
-  auto graph = new sycl::ext::oneapi::experimental::command_graph<
-      sycl::ext::oneapi::experimental::graph_state::modifiable>(
-      queue_ptr->get_context(), queue_ptr->get_device());
-  auto result = queue_graph_map.insert({queue_ptr, graph});
-  if (!result.second) {
-    return false;
+class graph_mgr {
```

Comment:
I prefer to put this class in the dpct::detail namespace.

Reply:
Done